app/models/miq_provision_virt_workflow.rb

```diff
@@ -1098,10 +1098,8 @@ def self.from_ws_ver_1_x(version, user, template_fields, vm_fields, requester, t
 
     init_options = {:use_pre_dialog => false, :request_type => request_type(parse_ws_string(template_fields)[:request_type])}
     data = parse_ws_string(requester)
-    unless data[:user_name].blank?
-      user = User.find_by_userid!(data[:user_name])
-      _log.warn "Web-service requester changed to <#{user.userid}>"
-    end
+
+    user = possibly_update_user(data, user)
```

Comment:
@syncrou Naming is hard.  Maybe `update_requester_from_parameters`

The new method can also be used from the automate_request: https://github.com/ManageIQ/manageiq/blob/ad4b906ae902e21c16c97958704fba4f5797b1f9/app/models/automation_request.rb#L20